agent/consul/leader_connect_test.go

```diff
@@ -1110,15 +1109,18 @@ func TestLeader_SecondaryCA_UpgradeBeforePrimary(t *testing.T) {
 		require.Equal(r, roots1[0].ID, roots2[0].ID)
 		require.Equal(r, roots1[0].RootCert, roots2[0].RootCert)
 
+		secondaryProvider, _ := getCAProviderWithLock(s2)
```

Comment:
This had to move into the retry loop, otherwise it was pointing at the old not initialized provider that existed before the secondary could be initialized.